tests/jerry/array-prototype-splice.js

```diff
@@ -175,3 +175,221 @@ try {
+  assert(e.message == "foo0");
+}
+
+// Checking behavior when a modified object is an element of the array
+try {
+  f = function() {delete arr[3]; arr.length = 13; Object.defineProperty(arr, '5', function() { })};
+  obj = {get: f, valueOf : f, toString: f};
+  arr = [1, 2, obj, 4, 5];
+  Object.defineProperty(arr, '2',{ 'get' : f } );
+  for(var i = 0; i < arr.length; i++)
+  {
+    var a = arr[i];
+  }
+  arr.splice(1, 4, obj);
+  assert(false);
+} catch (e) {
+  assert(e instanceof TypeError);
+}
+
+// Checking behavior when a modified object is an element of the array and deletes the elements
+try {
+  f = function() { for(var i = 0; i < arr.length; i++) { delete arr[i]} };
+  obj = {get: f, valueOf : f, toString: f};
+  arr = [1, 2, obj, 4, 5];
+  for(var i = 0; i < 6; i++)
+  {
+    Object.defineProperty(arr, i, { 'get' : f });
+  }
+  arr.splice(1, 3, obj);
+  assert(false);
+} catch (e) {
+  assert(e instanceof TypeError);
+}
+
+// Checking behavior when an element of the array throws error
+try {
+  f = function() { throw new TypeError("4");};
+  obj = {get: f, valueOf : f, toString: f};
+  arr = [1, 2, obj, 4, 5];
+  Object.defineProperty(arr, '4', { 'get' : f });
+  arr.splice(1, 3, obj);
+  assert(false);
+} catch (e) {
+  assert(e instanceof TypeError);
+  assert(e.message == "4");
+}
+
+// Checking behavior when elements are getting deleted the following way
+try{
+  f = function() { for(var i = 0; i < arr.length; i++) { delete arr[i]} };
+  f1 = function() { delete arr[2];};
+  f2 = function() { delete arr[3];};
+  f3 = function() { delete arr[4];};
+  arr = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+  delete arr[2];
+  delete arr[3];
+  Object.defineProperty(arr, '0', { 'get' : f });
+  Object.defineProperty(arr, '1', { 'get' : f });
+  Object.defineProperty(arr, '2', { 'get' : f });
+  Object.defineProperty(arr, '3', { 'get' : f });
+  Object.defineProperty(arr, '8', { 'get' : f1 });
+  Object.defineProperty(arr, '9', { 'get' : f2 });
+  Object.defineProperty(arr, '10', { 'get' : f3 });
+  arr.splice(1, 7, 5);
+} catch (e) {
+  assert(e instanceof TypeError);
+}
+
+// Checking behaior when elements are getting deleted the same way, but there are more elements and arguments
+try {
+  f = function() { for(var i=0; i<arr.length; i++) { delete arr[i]} };
+  f1 = function() { delete arr[2];};
+  f2 = function() { delete arr[3];};
+  f3 = function() { delete arr[4];};
+  arr = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24];
+  delete arr[2];
+  delete arr[3];
+  Object.defineProperty(arr, '0', { 'get' : f });
+  Object.defineProperty(arr, '1', { 'get' : f });
+  Object.defineProperty(arr, '2', { 'get' : f });
+  Object.defineProperty(arr, '3', { 'get' : f });
+  Object.defineProperty(arr, '8', { 'get' : f });
+  Object.defineProperty(arr, '9', { 'get' : f2 });
+  Object.defineProperty(arr, '10', { 'get' : f3 });
+  arr.splice(1, 7, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5);
```

Comment:
Is this really the simplest use case to test a feature of the engine? Is a 24 element array, 7 manually defined properties, and a 21 element splice needed to cover some path?